git-archive-version.cmake

```diff
@@ -0,0 +1 @@
+set(GIT_ARCHIVE_VERSION [=[$Format:%(describe)$]=])
```

Comment:
A small comment that this version is set by `git archive` wouldn't hurt (and phrasing it something like that would make it correct in both cases, i.e. when looking at the file with the placeholder and the file with the actual describe string).